### PR TITLE
Redescribe helper tests - Closes #930

### LINKED ISSUE #930: Refactor/Redescribe helpers tests
Recently, we had some helpers unit tests added by @reasonthearchitect . These test are pretty good, but we need to adjust the descriptions to match our own code standards. 

The comparison of all of the tests added can be found below.

https://github.com/LiskHQ/lisk/compare/1.0.0...1.0.0-Holding_Zone?expand=1

***

- [x] https://github.com/LiskHQ/lisk/blob/development/helpers/bignum.js
- [x] https://github.com/LiskHQ/lisk/blob/development/helpers/cache.js
- [x] https://github.com/LiskHQ/lisk/blob/development/helpers/checkIpInList.js
- [x] https://github.com/LiskHQ/lisk/blob/development/helpers/git.js
- [x] https://github.com/LiskHQ/lisk/blob/development/helpers/httpApi.js
- [x] https://github.com/LiskHQ/lisk/blob/development/helpers/inserts.js
- [x] https://github.com/LiskHQ/lisk/blob/development/helpers/promiseDefer.js
- [ ] https://github.com/LiskHQ/lisk/blob/development/helpers/router.js
- [x] https://github.com/LiskHQ/lisk/blob/development/helpers/z_schema-express.js

### PR DESCRIPTION
**Pull request by Tschakki**
Refactor / redescribe test descriptions for helpers

Closes #930

### CODE REVIEW

File: test/unit/helpers/git.js [hunk 1]
```diff
@@ -6,8 +6,11 @@ var sinon = require('sinon');
 var git = require('../../../helpers/git');
 var childProcess = require('child_process');
 
-describe('git', function () {
-	describe('has an error', function () {
+describe('getLastCommit', function () {
```
Comment (Isabello) [inline test/unit/helpers/git.js]: We should describe with git, then encapsulate the rest inside of this `getLastCommit` describe block

File: test/unit/helpers/inserts.js [hunk 2]
```diff
@@ -8,49 +8,64 @@ var inserts = require('../../../helpers/inserts');
...
+
+		it('should throw error');
+
+		standardExpect(function () {new inserts({table: 'atable', fields: []}, null, null);}, 'values');
 	});
 
 	describe('validate the template function', function () {
-		it('concatinates  = false with a single argument', function () {
+
+		describe('when concatinates  = false with a single argument', function () {
```
Comment (Isabello) [inline test/unit/helpers/inserts.js]: concatenates

Comment (Isabello) [reply test/unit/helpers/inserts.js]: next few describes the same applies

File: test/unit/helpers/promiseDefer.js [hunk 3]
```diff
@@ -21,13 +21,15 @@ describe('PromiseDefer', function () {
 		});
 	});
 
-	it('resolves', function (done) {
+	it('should resolve', function (done) {
```
Comment (MaciejBaj) [inline test/unit/helpers/promiseDefer.js]: Tests need to be refactored as the descriptions are not really helpful 
```
describe('PromiseDefer')
   it('should return result containing resolve)
   it('should return result containing reject)
   it('should return result containing promise)
   it('promise.isFullfilled should return false');
   describe('when promiseDefer fails')
      it('promise.isFullfilled should return true');
      it('should call prosise.catch');
   describe('when promiseDefer succeeds')
      it('promise.isFullfilled should return true');
      it('should call prosise.then');
```

File: test/unit/helpers/inserts.js [hunk 4]
```diff
@@ -8,49 +8,64 @@ var inserts = require('../../../helpers/inserts');
```
Comment (MaciejBaj) [inline test/unit/helpers/inserts.js]: Please merge the tests with expectations

File: test/unit/helpers/httpApi.js [hunk 5]
```diff
@@ -171,39 +171,46 @@ describe('httpApi', function () {
```
Comment (MaciejBaj) [inline test/unit/helpers/httpApi.js]: These tests should be either removed or refactor.

**Comment by karmacoma** [comment]
Closing this pull request. We will divide the work into individual pull requests per helper.

**Pull request closed without merging**